### ISSUE TITLE
is there a WYSIWYG editor?

### ISSUE THREAD

**Issue by Jp3rd** [Gottwik/Enduro#71]
or is there a workaround?

**Comment by CosmoMyzrailGorynych** [comment]
What do you mean? Text editor, markddown, theme editor, etc?

**Comment by Gottwik** [comment]
There is no wysiwyg in enduro yet. What I do when I need a formatted text is have a textarea with a markdown abstractor(abstractor converts the markdown to html).

If you believe there is a use case for wysiwyg, comment below or submit a pull request.

**Comment by mdxprograms** [comment]
is it possible to "hook" into the admin with a wysiwyg plugin? just set a type for something like below? Then use something like [showdown](https://github.com/showdownjs/showdown) for markdown parsing on the frontend
```javascript
{
  $content_type: 'editor',
  content: '## my heading goes here'
}
```

**Comment by CosmoMyzrailGorynych** [comment]
_/* Pell is not a markdown editor, isn't it? */_

It is definitely posssible to hook up a UI script on a front side and make an editor. We only need a special input type to  define such fields.

I will try to make such with SimpleMDE tonight, 'cause it is really a simple but needed task.

**Comment by mdxprograms** [comment]
That would be awesome. I'm always willing to contribute as well 👍🏼

Ya I realized Pell wasn't the best example but it would be cool if there was a markdown editor none the less. 

Really like this project. Please let me know if you guys need anything. 

_Sent from my Google Pixel using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by CosmoMyzrailGorynych** [comment]
Nah, I can't render enduro_admin UI because of strange error and definitely won't be able to tie [SimpleMDE](https://simplemde.com/) to it in nearest future :|

**Comment by mdxprograms** [comment]
What's the error? I can help take a look

_Sent from my Google Pixel using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Gottwik** [comment]
@mdxprograms: super interesting stuff. It's even simple, because pell outputs html so you can just store that without any markdown conversion going on.

![jul-16-2017 03-02-59](https://user-images.githubusercontent.com/277311/28243699-10d817be-69d5-11e7-807b-854f1bce6132.gif)

made a sample for you: https://github.com/Gottwik/enduro_samples/tree/master/wysiwyg

Let me know if it is what you were looking for. I will work on documenting these features better.

**Comment by mdxprograms** [comment]
checking it out now

**Comment by CosmoMyzrailGorynych** [comment]
@mdxprograms it is here: https://github.com/Gottwik/enduro_admin/issues/12
It seems that edited copy of enduro_admin is not linked to enduro, though I did everything said in instructions. And there is also a strange error while rendering index.hbs.

**Comment by mdxprograms** [comment]
hmm @Gottwik @CosmoMyzrailGorynych I'm getting the following and the textbox is only showing the raw markup. @CosmoMyzrailGorynych is this related to https://github.com/Gottwik/enduro_admin/issues/12?

![screen shot 2017-07-16 at 12 42 36 am](https://user-images.githubusercontent.com/1900735/28244715-0c7466c4-69c0-11e7-8c6e-4f1f2296fa13.png)
![screen shot 2017-07-16 at 12 44 22 am](https://user-images.githubusercontent.com/1900735/28244716-0c764ce6-69c0-11e7-848c-088e0ddf5915.png)

**Comment by CosmoMyzrailGorynych** [comment]
@mdxprograms  idk but am trying to make similar trick with SimpleMDE. I get exactly the same error because admin extensions can't see `enduro_admin_app.compileProvider`.

**Comment by mdxprograms** [comment]
hmm is it possible to use the CDN on a global scale just to see if that would fix it?

**Comment by Gottwik** [comment]
ehm, my bad, you have to **update enduro**(npm i enduro -g) :-/ I added the compileProvider for this thing to work.

**Comment by CosmoMyzrailGorynych** [comment]
Just done this but it didn't help

**Comment by Gottwik** [comment]
do you still see the compileProvider error? Or is it the invalid regular expression thing?

**Comment by CosmoMyzrailGorynych** [comment]
It is still a `.compileProvider` error. My enduro version is 1.4.30; maybe npm needs some time to refresh its cache, idk?

**Comment by Gottwik** [comment]
Hey, the problem could be that you have cloned the enduro_admin and used npm link.

Maybe just git pull your enduro_admin and this will solve itself.

**Comment by CosmoMyzrailGorynych** [comment]
Still doesn't work.
My repo is here:

![image](https://user-images.githubusercontent.com/3174357/28245150-2ad850e6-6a54-11e7-8106-2178e3600a86.png)

I did `npm link` in enduro_admin` folder and `npm link enduro_admin` in `C:\Users\Космо\AppData\Roaming\npm\node_modules\enduro>`.

**Comment by Gottwik** [comment]
Either you figure out the npm linking or you just pull enduro and run `npm i`. That will fetch the newest enduro_admin

I think you need to do `npm link` in enduro folder. This sound the `npm linking` got screwed up(might be because of the poor instructions on enduro_admin).

How does it look when you go to `<folder where you have enduro cloned> /node_modules/enduro_admin/package.json`. You should be seeing version 1.0.64.

**Comment by CosmoMyzrailGorynych** [comment]
Me:
![help](https://media.giphy.com/media/rAm0u2k17rM3e/giphy.gif)

**Enduro doesn't install bower dependencies on Windows. It launches `node_modules/bower/bin/bower install` instead of `./node_modules/bower/bin/bower install` and therefore fails. Now I have a non-working global copy of enduro.

**Comment by CosmoMyzrailGorynych** [comment]
@Gottwik how can we listen to culture switching?

**Comment by Gottwik** [comment]
¯\_(ツ)_/¯

what does bower has to do with enduro? Themes?

I guess the linking is messed up. Either reinstall everything(I use nvm, which makes this easier I believe) or try to figure out the linking with https://docs.npmjs.com/cli/ls#link

Hey, let me know if I can help. I am happy to follow your steps to see if I get different results.

**Comment by CosmoMyzrailGorynych** [comment]
@Gottwik I have already solved linking problems and even created a SimpleMDE field. Looks like you were seeing an old thread from email or so O.o

**Comment by Gottwik** [comment]
@CosmoMyzrailGorynych You can add this to make it work with cultures. I will add it to the sample.

```
scope.$watch('current_culture', function () {
    current_content.content.innerHTML = scope.context[scope.terminatedkey] || ''
})
```

@Jp3rd Were we able to answer your question?

@mdxprograms Does this work for you? Did you expect more packaged solution or is this fine?

**Comment by CosmoMyzrailGorynych** [comment]
Wheee, it works ^—^
I will add a sample project soon; here's a quick setup:

1. run `bower install simplemde --save`  in your project directory;
2. create two files:  
[\assets\admin_extensions\marked.js](https://gist.github.com/CosmoMyzrailGorynych/e61dfc0a51a564631670af199311ad8c)  
[\pages\custom_types\marked.hbs](https://gist.github.com/CosmoMyzrailGorynych/854d9dab4dc9744b2a9bd5fae73557c3)

@Gottwik I can write documentation/tutorial for endurojs.com if you want.

**Comment by Ezra-Siton-UIX** [comment]
Hi. This feature is so basic and usefull (clients who manage the site **can not write MD markup**!!) - why this feature is not build in - in enduro?
